### PR TITLE
UserContextValues's userRole in App now uses backend api to get user's role

### PR DESCRIPTION
**Pull request by thaigillespie**
It used to get the roles from firebase custom claims.

### CODE REVIEW

**Comment by thaigillespie** [comment]
Lmao since this needs the auth page refresh bug fix + legit test accounts the cypress literally cannot pass : (

**Comment by thaigillespie** [comment]
Or well it needs the TODOs that I put in as comments at least. I left the TODOs there bc I thouht you'd have to do smth like that for the auth page refresh bug fix @godwinpang. Let me know if that is or isn't the case. If it isn't, I'll do the TODOs myself.

File: src/pages/App/index.tsx [hunk 1]
```diff
@@ -38,9 +38,14 @@ function App(): JSX.Element {
         const tokenResult = await user.getIdTokenResult();
         const { claims, token } = tokenResult;
 
+        // TODO: Add ApiConfigStore.setToken(token || '') here so getUserRole() works
+
+        const id = parseInt(claims.user_id, 10);
+        const userRole = await getUserRole(id);
```
Comment (godwinpang) [inline src/pages/App/index.tsx:44]: don't await on both here - make them promises and do a promise.all - i want both awaits to go out together

Comment (godwinpang) [inline src/pages/App/index.tsx:44]: don't await on both here - make them promises and do a promise.all - i want both awaits to go out together

**Review by godwinpang — CHANGES_REQUESTED** [review]

**Review by godwinpang — APPROVED** [review]

**Pull request merged**